src/game_config.cpp

```diff
@@ -0,0 +1,171 @@
+	const auto& path = (arg_path.empty()) ? default_path : arg_path;
+
+	if (!path.empty()) {
+		cfg.loadFromConfig(path);
+	}
+
+	cp.Rewind();
+	cfg.loadFromArgs(cp);
+
+	return cfg;
+}
+
+std::string Game_Config::GetDefaultConfigPath() {
+	// FIXME: Platform specific add this.
+	// FIXME: Game specific configs?
+
+	return "";
+}
+
+std::string Game_Config::GetConfigPath(CmdlineParser& cp) {
+	std::string path;
+
+	while (!cp.Done()) {
+		CmdlineArg arg;
+		if (cp.ParseNext(arg, 1, "--config", 'c')) {
+			if (arg.NumValues() > 0) {
+				path = arg.Value(0);
+			}
+			continue;
+		}
+
+		cp.SkipNext();
+	}
+
+	return path;
+}
+
+void Game_Config::loadFromArgs(CmdlineParser& cp) {
+	while (!cp.Done()) {
+		CmdlineArg arg;
+		long li_value = 0;
+		if (cp.ParseNext(arg, 0, "--vsync")) {
+			video.vsync.Set(true);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--no-vsync")) {
+			video.vsync.Set(false);
+			continue;
+		}
+		if (cp.ParseNext(arg, 1, "--fps-limit")) {
+			if (arg.ParseValue(0, li_value)) {
+				video.fps_limit.Set(li_value);
+			}
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--no-fps-limit")) {
+			video.fps_limit.Set(0);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--show-fps")) {
+			video.show_fps.Set(true);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--no-show-fps")) {
+			video.show_fps.Set(false);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--fps-render-window")) {
+			video.fps_render_window.Set(true);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--no-fps-render-window")) {
+			video.fps_render_window.Set(false);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--no-show-fps")) {
+			video.show_fps.Set(false);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--window")) {
+			video.fullscreen.Set(false);
+			continue;
+		}
+		if (cp.ParseNext(arg, 0, "--fullscreen")) {
+			video.fullscreen.Set(true);
+			continue;
+		}
+		if (cp.ParseNext(arg, 1, "--window-zoom")) {
+			if (arg.ParseValue(0, li_value)) {
+				video.window_zoom.Set(li_value);
+			}
+			continue;
+		}
+
+		cp.SkipNext();
+	}
+}
+
+void Game_Config::loadFromConfig(const std::string& path) {
+	this->config_path = path;
+
+	lcf::INIReader ini(path);
+
+	if (ini.ParseError()) {
+		Output::Debug("Failed to parse ini config file {}", path);
+		return;
+	}
+
+	if (ini.HasValue("video", "vsync")) {
+		video.vsync.Set(ini.GetBoolean("video", "vsync", false));
+	}
+	if (ini.HasValue("video", "fullscreen")) {
+		video.vsync.Set(ini.GetBoolean("video", "fullscreen", false));
+	}
+	if (ini.HasValue("video", "show-fps")) {
+		video.vsync.Set(ini.GetBoolean("video", "show-fps", false));
+	}
+	if (ini.HasValue("video", "fps-render-window")) {
+		video.vsync.Set(ini.GetBoolean("video", "fps-render-window", false));
+	}
+	if (ini.HasValue("video", "fps-limit")) {
+		video.vsync.Set(ini.GetInteger("video", "fps-limit", 0));
+	}
+	if (ini.HasValue("video", "window-zoom")) {
+		video.vsync.Set(ini.GetInteger("video", "window-zoom", 0));
```

Comment:
This all sets the wrong field ``video.vsync`` intentional because WIP (add comment :)) or a bug?

Reply:
Thats a bug for sure. Fixed!